doc/dev/tests.md

```diff
@@ -264,32 +264,41 @@ From your terminal run the `pytest` command to run all the tests that you have w
 
 Your update should run smooth and have green dots representing passing tests. Now if you look at the contents of your `tests` directory there should be a new directory called `recording` with four `.yaml` files. Each `yaml` file is a recording for a single test. To run a test in playback mode change the `testsettings_local.cfg` to `live-mode: false` and rerun the tests with the same command. The test infrastructure will use the automatically created `.yaml` recordings to mock the HTTP traffic and run the tests.
 
+## Functional vs. Unit Tests
 
-## More Test Examples
-
-This section will demonstrate how to write tests with the `devtools_testutils` package with a few samples to showcase the features of the test framework.
-
-### Example 1: Basic Recording and Interactions
+The test written above is a functional test, it generates HTTP traffic and sends data to the service. Most of our clients have some client-side validation for account names, formatting, or properties that do not generate HTTP traffic. For unit tests, the best practice is to have a separate test class from the `AzureTestCase` class which tests client side validation methods. For example, the `azure-data-tables` library has client-side validation for the table name and properties of the entity, below is an example of how these could be tested:
```

Comment:
not really necessary, but might be useful to link to [`AzureTestCase`](https://github.com/Azure/azure-sdk-for-python/blob/ce3bfd5b961287dca2c2eb4769d428eec0585c0d/tools/azure-sdk-tools/devtools_testutils/azure_testcase.py) file here, or wherever it's first mentioned?